agent.go

```diff
@@ -468,8 +468,12 @@ func (n *network) addDriverWatches() {
 
 	c := n.getController()
 	for _, tableName := range n.driverTables {
-		ch, cancel := c.agent.networkDB.Watch(tableName, n.ID(), "")
 		c.Lock()
+		if c.agent == nil {
```

Comment:
You need to retrieve agent within lock and only use that after. Not `c.agent` anymore.